src/ui/public/embeddable/container_api.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * The ContainerAPI is an interface for embeddable objects to interact with the container they are embedded within.
+ */
+export class ContainerAPI {
+  /**
+   * Available so the embeddable object can trigger a filter action.
+   * @param field
+   * @param value
+   * @param operator
+   * @param index
+   */
+  onFilter(/*field, value, operator, index */) {
+    throw new Error('Must implement onFilter.');
+  }
+
+  /**
+   * @return {AppState}
+   */
+  getAppState() {
+    throw new Error('Must implement getAppState.');
+  }
+
+  /**
+   * @return {boolean} If the container is marked as being in view only mode or not.
+   */
+  getIsViewOnlyMode() {
+    throw new Error('Must implement getIsViewOnlyMode.');
+  }
+
+  /**
+   * Creates a child ui state for the panel. It's passed the ui state to use, but needs to
+   * be generated from the parent.
+   * @param path {String} - the unique path for this ui state.
+   * @param uiState {Object} - the uiState for the child.
+   * @returns {Object}
+   */
+  createChildUiState(/* path, uiState */) {
```

Comment:
Can we call this `initalState` or something that disambiguates it from the `PersistedState` instances we usually call `uiState`? If it was a `PersistedState` then we would just be able to call `createChild()` on it.